aspnetcore/migration/22-to-30.md

```diff
@@ -437,35 +437,51 @@ The most significant change from `WebHostBuilder` to `HostBuilder` is in [depend
 
 ## Update SignalR code
 
-If you call `AddJsonProtocol`, replace it with `AddNewtonsoftJsonProtocol`.
+`System.Text.Json` is now the default Hub Protocol used by both the client and server.
 
-* The following examples show server code before and after the change:
+In `Startup.ConfigureServices`, call `AddJsonProtocol` to set serializer options.
 
-  ```csharp
-  services.AddSignalR(...)
-          .AddJsonProtocol(...) // 2.2
-  ```
+**Server:**
 
-  ```csharp
-  services.AddSignalR(...)
-          .AddNewtonsoftJsonProtocol(...) // 3.0
-  ```
+```csharp
+services.AddSignalR(...)
+        .AddJsonProtocol(options =>
+        {
+            options.WriteIndented = false;
+        })
+```
 
-* The following examples show .NET client code before and after the change:
+**Client:**
 
-  ```csharp
-  connection = new HubConnectionBuilder()
-      .WithUrl(...)
-      .AddJsonProtocol(...) // 2.2
-      .Build()
-  ```
+```csharp
+new HubConnectionBuilder()
+    .WithUrl("/chatHub")
+    .AddJsonProtocol(options =>
+    {
+        options.WriteIndented = false;
+    })
+    .Build();
+```
 
-  ```csharp
-  connection = new HubConnectionBuilder()
-      .WithUrl(...)
-      .AddNewtonsoftJsonProtocol(...) // 3.0
-      .Build()
-  ```
+### Switch to Newtonsoft.Json
+If you're using features of `Newtonsoft.Json` that aren't supported in `System.Text.Json`, you can switch back to `Newtonsoft.Json`:
+
+1. Install the [Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson](https://www.nuget.org/packages/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson) NuGet package.
+1. On the client, chain an `AddNewtonsoftJsonProtocol` method call to the `HubConnectionBuilder` instance:
+
+```csharp
+new HubConnectionBuilder()
+    .WithUrl("/chatHub")
+    .AddNewtonsoftJsonProtocol(...)
+    .Build();
+```
+
+3. On the server, chain an `AddNewtonsoftJsonProtocol` method call to the `AddSignalR` method call in `Startup.ConfigureServices`:
```

Comment:
```suggestion
1. On the server, chain an `AddNewtonsoftJsonProtocol` method call to the `AddSignalR` method call in `Startup.ConfigureServices`:
```

Reply:
This wasn't working before because of the code snippet. Maybe it works now that there is a newline?

Reply:
Nope, reverted the change